### ISSUE TITLE
How to reshape and slice a variable without converting it to an untrainable tensor?

### ISSUE THREAD

**Issue by yashkant** [tensorflow/tensorflow#21844]
[Feature Request]

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**:
- **TensorFlow installed from (source or binary)**: pip-install
- **TensorFlow version (use command below)**: v1.10.0-0-g656e7a2b34 1.10.0
- **Python version**: Python 2.7.12
- **Bazel version (if compiling from source)**: 
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**: Cuda compilation tools, release 7.5, V7.5.17
- **GPU model and memory**: TITAN Xp 
- **Exact command to reproduce**:  Described below 

### Describe the problem

I am using custom trainable variables in my layers, I pass the variables as a parameter to the layer (which are custom too).

I want to set (or pass) only a slice of my trainable variables array to the layer. 

For example: 

    custom_train_vars = tf.get_variable(name, [10,30,60,60], initializer=initializer, trainable=trainable)
    layer_weight = custom_train_vars[2]
    x = my_custom_conv_layer(x, layer_weight, args..)

The problem is after slicing (or reshaping with `tf.reshape`) the variable i.e. `layer_weight = custom_train_vars[2]`, it gets converted to the type `Tensor("strided_slice:0",args..)`

And I'm unable to use this as a training variable inside my custom layer, because my optimizer throws an error while finding the gradient of this Tensor. 

For example:

    Optimizer.compute_gradients(train_loss, layer_weight)

 Is there a workaround/feature where the slicing won't change the type of variable?

**Comment by yashkant** [comment]
@tatatodd  Could you please review this issue? :)

**Comment by ymodak** [comment]
@yashkant Can you please post the error the message you are getting?

**Comment by ymodak** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
